darwin/utils.py

```diff
@@ -312,129 +290,131 @@ def parse_darwin_json(path: Path, count: Optional[int]) -> Optional[dt.Annotatio
+    return annotation_file
 
 
-def _parse_darwin_image(path: Path, data: Dict[str, Any], count: Optional[int]) -> dt.AnnotationFile:
-    annotations: List[dt.Annotation] = list(filter(None, map(_parse_darwin_annotation, data["annotations"])))
-    annotation_classes: Set[dt.AnnotationClass] = set([annotation.annotation_class for annotation in annotations])
-    return dt.AnnotationFile(
-        path,
-        _get_local_filename(data["image"]),
-        annotation_classes,
-        annotations,
-        False,
-        data["image"].get("width"),
-        data["image"].get("height"),
-        data["image"].get("url"),
-        data["image"].get("workview_url"),
-        data["image"].get("seq", count),
-        None,
-        data["image"].get("path", "/"),
+def _parse_darwin_slot(data: Dict[str, Any]) -> dt.Slot:
+    return dt.Slot(
+        name=data["slot_name"],
+        type=data["type"],
+        width=data.get("width"),
+        height=data.get("height"),
+        source_files=data.get("source_files", []),
+        thubmnail_url=data.get("thumbnail_url"),
+        frame_count=data.get("frame_count"),
+        frame_urls=data.get("frame_urls"),
+        fps=data.get("fps"),
     )
 
 
-@deprecation.deprecated(
-    deprecated_in="0.7.5",
-    removed_in="0.8.0",
-    current_version=__version__,
-    details="Use 'parse_darwin_json' instead",
-)
-def parse_darwin_video(path: Path, data: Dict[str, Any], count: Optional[int]) -> dt.AnnotationFile:
-    """
-    Parses the given JSON file in v7's darwin proprietary format. Works for playback videos.
-
-    Parameters
-    ----------
-    path : Path
-        Path to the file to parse.
-    data : Dict[str, Any]
-        The decoded JSON file in Python format.
-    count : Optional[int]
-        Optional count parameter. Used only if the data["image"]["seq"] sequence is None.
-
-    Returns
-    -------
-    dt.AnnotationFile
-        An AnnotationFile with the information from the parsed JSON file.
-    """
-
-    annotations: List[dt.VideoAnnotation] = list(filter(None, map(parse_darwin_video_annotation, data["annotations"])))
+def _parse_darwin_image(path: Path, data: Dict[str, Any], count: Optional[int]) -> dt.AnnotationFile:
+    annotations: List[Union[dt.Annotation, dt.VideoAnnotation]] = _data_to_annotations(data)
     annotation_classes: Set[dt.AnnotationClass] = set([annotation.annotation_class for annotation in annotations])
 
-    if "width" not in data["image"] or "height" not in data["image"]:
-        raise OutdatedDarwinJSONFormat("Missing width/height in video, please re-export")
+    slot = dt.Slot(
+        name=None,
+        type="image",
+        source_files=[{"url": data["image"].get("url"), "file_name": _get_local_filename(data["image"])}],
+        thubmnail_url=data["image"].get("thubmnail_url"),
+        width=data["image"].get("width"),
+        height=data["image"].get("height"),
+    )
 
-    return dt.AnnotationFile(
+    annotation_file = dt.AnnotationFile(
```

Comment:
This is not an action point for this PR but using named arguments would make these AnnotationFile instantiations much easier to read.

Reply:
True, but yes, I don't want to put more time into this PR right now.